### PR TITLE
feat: add HCP Vault Secrets API docs

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

This PR adds API documentation for HCP Vault Secrets at [/hcp/api-docs/vault-secrets].

## 🤷 Why

To better expose documentation for this API to users.

## 🛠️ How

Applies existing work with HCP Packer API docs to the analogous Swagger file for HCP Vault Secrets.

## 📸 Design Screenshots

> **Note**: this is a net new page

localhost_3000_hcp_api-docs_vault-secrets

## 🧪 Testing

- [ ] Visit [/hcp/api-docs/vault-secrets]
    - The page should look and function as expected

## 💭 Anything else?

- For now, we've stuck to the existing API docs template, without further modifications
- Intent is to favour rebuilding the API docs template from scratch over tweaking the current template

[task]: https://app.asana.com/0/1204810348326275/1204973521875671/f
[preview]: https://dev-portal-git-zshcp-vault-secrets-api-docs-hashicorp.vercel.app/
[/hcp/api-docs/vault-secrets]: https://dev-portal-git-zshcp-vault-secrets-api-docs-hashicorp.vercel.app/hcp/api-docs/vault-secrets

### CODE REVIEW

**Review by emilypersson1 — APPROVED** [review]
nice, thank you!

**Comment by kartiklunkad26** [comment]
@zchsh working on updating the title & `tags/name` but wanted to ask about the `tags/name`. What is the purpose of the sub-heading? 

Is the intent that multiple sub-sections can exist? For the `tags/name`, honestly we wouldn't want a sub-heading at all

**Comment by zchsh** [comment]
@kartiklunkad26 Great question! The intent behind showing that subheading is to display the service name, which made sense in the context of the Boundary API docs for which this template was originally built (see https://developer.hashicorp.com/boundary/api-docs/account-service)... but obviously isn't quite as applicable in this case!

If we'd like, I can remove the `Secret Service` heading as a one-off modification for the HCP Vault Secrets API docs 👍

**Comment by kartiklunkad26** [comment]
Perfect!

**Comment by zchsh** [comment]
@kartiklunkad26 I've update this PR to remove the subheading, and updated the screenshot in the PR description. Should take just a couple minutes for the [preview](https://dev-portal-git-zshcp-vault-secrets-api-docs-hashicorp.vercel.app/hcp/api-docs/vault-secrets) to rebuild!

File: src/views/api-docs-view/components/path-truncation-aside/index.tsx [hunk 1]
```diff
@@ -6,6 +6,17 @@
 import DevDotContent from 'components/dev-dot-content'
 import CodeBlock from '@hashicorp/react-code-block'
 
+/**
+ * Truncates HCP Vault Secrets API operation paths for clarity.
+ * Intended to be used with a `<PathTruncationAside />` in each operation intro.
+ */
+function truncateVaultSecretsOperationPath(path: string) {
+	return path.replace(
```
Comment (kendallstrautman) [inline src/views/api-docs-view/components/path-truncation-aside/index.tsx:14]: ⛏️ Could add add a bit more clarity as to what the regex should match on in the comment?

File: src/views/api-docs-view/index.tsx [hunk 2]
```diff
@@ -36,12 +36,12 @@ function ApiDocsView({
...
 	 * We always render the API docs name in a heading-styled element.
 	 *
-	 * When `serviceData` is provided, we'll render the service name
+	 * When `serviceData.name` is provided, we'll render the service name
 	 * in an `h1` element, as the `serviceName` is a more meaningful page title.
 	 * In such cases, our page heading needs to be a `p` element to avoid
 	 * having multiple `h1` elements on the page.
 	 */
-	const pageHeadingTag = serviceData ? 'p' : 'h1'
+	const pageHeadingTag = serviceData && serviceData.name ? 'p' : 'h1'
```
Comment (kendallstrautman) [inline src/views/api-docs-view/index.tsx:44]:
```suggestion
	const pageHeadingTag = serviceData?.name ? 'p' : 'h1'
```
💭 Just thinking out-loud, either way is fine.

**Review by kendallstrautman — APPROVED** [review]
LGTM!

File: src/views/api-docs-view/components/path-truncation-aside/index.tsx (hunk 1 above)
Comment (zchsh) [reply src/views/api-docs-view/components/path-truncation-aside/index.tsx:18]: 🙌 Great call! Added this in 👍

**Pull request merged**